plasmapy/utils/decorators/checks.py

```diff
@@ -302,6 +304,7 @@ def _check_value(self, arg, arg_name: str, arg_checks: Dict[str, bool]):
 
         # check values
         # * 'none_shall_pass' always needs to be checked first
+        # * 'can_be_nan' should be second to simplify other checks
```

Comment:
```suggestion
```

I don't think this order is necessary.  I'm actually not sure why the `'can_be_negative'` condition was written like it was.  I'm assuming a previous version of `numpy` did not handle `np.nan` and `np.inf` properly in conditional expressions, but that's just a guess.

Reply:
I think it would be ok to simplify `can_be_negative condition` (remove context manager `with np.errstate`). If you suggestion is right, then it should be removed. Nevertheless, no failed tests with nans in CheckValues test suite are found.